### PR TITLE
ASoC: SOF: Intel: fix kconfig related if statements in machine drivers

### PR DESCRIPTION
**Pull request by xun2z**
Use SND_SOC_SOF_INTEL_TOPLEVEL configuration in some if statements of
machine drivers (SND_SOC_SOF_INTEL no longer exists)

This is to fix issue #474 , which blocks SOF from loading topology file.

### CODE REVIEW

File: sound/soc/intel/boards/bytcht_da7213.c [hunk 1]
```diff
@@ -43,6 +43,10 @@ static const struct snd_soc_dapm_widget dapm_widgets[] = {
 	SND_SOC_DAPM_MIC("Headset Mic", NULL),
 	SND_SOC_DAPM_MIC("Mic", NULL),
 	SND_SOC_DAPM_LINE("Aux In", NULL),
+	SST_AIF_IN("codec_in0", NULL),
+	SST_AIF_IN("codec_in1", NULL),
+	SST_AIF_OUT("codec_out0", NULL),
+	SST_AIF_OUT("codec_out1", NULL),
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcht_da7213.c]: NAK? It looks like a complete misunderstanding, the ask was to create virtual widgets in the topology file e.g. https://github.com/thesofproject/sof/blob/a2e6f82ef810a6a838783467472dfceb7933afc1/tools/topology/sof-apl-da7219.m4#L174

File: sound/soc/intel/boards/bytcht_da7213.c [hunk 2]
```diff
@@ -56,7 +60,6 @@ static const struct snd_soc_dapm_route audio_map[] = {
 	{"MIC1", NULL, "Headset Mic"},
 	{"MIC2", NULL, "Mic"},
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcht_da7213.c]: that change is fine

File: sound/soc/intel/boards/bytcr_rt5640.c [hunk 3]
```diff
@@ -845,7 +850,6 @@ static int byt_rt5640_init(struct snd_soc_pcm_runtime *runtime)
 	if (ret)
 		return ret;
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcr_rt5640.c]: this is fine

File: sound/soc/intel/boards/bytcr_rt5651.c [hunk 4]
```diff
@@ -243,6 +243,12 @@ static const struct snd_soc_dapm_widget byt_rt5651_widgets[] = {
 	SND_SOC_DAPM_SUPPLY("Ext Amp Power", SND_SOC_NOPM, 0, 0,
 			    rt5651_ext_amp_power_event,
 			    SND_SOC_DAPM_PRE_PMD | SND_SOC_DAPM_POST_PMU),
+	SST_AIF_IN("modem_in", NULL),
+	SST_AIF_IN("codec_in0", NULL),
+	SST_AIF_IN("codec_in1", NULL),
+	SST_AIF_OUT("modem_out", NULL),
+	SST_AIF_OUT("codec_out0", NULL),
+	SST_AIF_OUT("codec_out1", NULL),
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcr_rt5651.c]: NAK

File: sound/soc/intel/boards/bytcr_rt5651.c [hunk 5]
```diff
@@ -974,7 +975,6 @@ static int snd_byt_rt5651_mc_probe(struct platform_device *pdev)
 			byt_rt5651_quirk |= BYT_RT5651_SSP0_AIF2;
 		}
 	}
-#endif
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcr_rt5651.c:977]: this is fine

File: sound/soc/intel/boards/cht_bsw_max98090_ti.c [hunk 6]
```diff
@@ -81,6 +81,10 @@ static const struct snd_soc_dapm_widget cht_dapm_widgets[] = {
 	SND_SOC_DAPM_SUPPLY("Platform Clock", SND_SOC_NOPM, 0, 0,
 			    platform_clock_control, SND_SOC_DAPM_PRE_PMU |
 			    SND_SOC_DAPM_POST_PMD),
+	SST_AIF_IN("codec_in0", NULL),
+	SST_AIF_IN("codec_in1", NULL),
+	SST_AIF_OUT("codec_out0", NULL),
+	SST_AIF_OUT("codec_out1", NULL),
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_max98090_ti.c]: nak

File: sound/soc/intel/boards/cht_bsw_max98090_ti.c [hunk 7]
```diff
@@ -91,14 +95,12 @@ static const struct snd_soc_dapm_route cht_audio_map[] = {
...
 	{"Ext Spk", NULL, "SPKL"},
 	{"Ext Spk", NULL, "SPKR"},
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
 	{"HiFi Playback", NULL, "ssp2 Tx"},
 	{"ssp2 Tx", NULL, "codec_out0"},
 	{"ssp2 Tx", NULL, "codec_out1"},
 	{"codec_in0", NULL, "ssp2 Rx" },
 	{"codec_in1", NULL, "ssp2 Rx" },
 	{"ssp2 Rx", NULL, "HiFi Capture"},
-#endif
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_max98090_ti.c:101]: ok

File: sound/soc/intel/boards/cht_bsw_rt5645.c [hunk 8]
```diff
@@ -121,6 +121,12 @@ static const struct snd_soc_dapm_widget cht_dapm_widgets[] = {
 	SND_SOC_DAPM_SPK("Ext Spk", NULL),
 	SND_SOC_DAPM_SUPPLY("Platform Clock", SND_SOC_NOPM, 0, 0,
 			platform_clock_control, SND_SOC_DAPM_PRE_PMU | SND_SOC_DAPM_POST_PMD),
+	SST_AIF_IN("modem_in", NULL),
+	SST_AIF_IN("codec_in0", NULL),
+	SST_AIF_IN("codec_in1", NULL),
+	SST_AIF_OUT("modem_out", NULL),
+	SST_AIF_OUT("codec_out0", NULL),
+	SST_AIF_OUT("codec_out1", NULL),
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_rt5645.c]: nak

File: sound/soc/intel/boards/cht_bsw_rt5645.c [hunk 9]
```diff
@@ -159,43 +165,35 @@ static const struct snd_soc_dapm_route cht_rt5650_audio_map[] = {
...
-#endif
 };
 
 static const struct snd_soc_dapm_route cht_rt5645_ssp0_aif2_map[] = {
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
 	{"AIF2 Playback", NULL, "ssp0 Tx"},
 	{"ssp0 Tx", NULL, "modem_out"},
 	{"modem_in", NULL, "ssp0 Rx" },
 	{"ssp0 Rx", NULL, "AIF2 Capture"},
-#endif
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_rt5645.c:198]: ok

File: sound/soc/intel/boards/cht_bsw_rt5672.c [hunk 10]
```diff
@@ -115,6 +115,10 @@ static const struct snd_soc_dapm_widget cht_dapm_widgets[] = {
 	SND_SOC_DAPM_SUPPLY("Platform Clock", SND_SOC_NOPM, 0, 0,
 			platform_clock_control, SND_SOC_DAPM_PRE_PMU |
 			SND_SOC_DAPM_POST_PMD),
+	SST_AIF_IN("codec_in0", NULL),
+	SST_AIF_IN("codec_in1", NULL),
+	SST_AIF_OUT("codec_out0", NULL),
+	SST_AIF_OUT("codec_out1", NULL),
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_rt5672.c]: nak

File: sound/soc/intel/boards/cht_bsw_rt5672.c [hunk 11]
```diff
@@ -128,14 +132,12 @@ static const struct snd_soc_dapm_route cht_audio_map[] = {
...
 	{"Ext Spk", NULL, "SPORP"},
 	{"Ext Spk", NULL, "SPORN"},
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
 	{"AIF1 Playback", NULL, "ssp2 Tx"},
 	{"ssp2 Tx", NULL, "codec_out0"},
 	{"ssp2 Tx", NULL, "codec_out1"},
 	{"codec_in0", NULL, "ssp2 Rx"},
 	{"codec_in1", NULL, "ssp2 Rx"},
 	{"ssp2 Rx", NULL, "AIF1 Capture"},
-#endif
```
Comment (plbossart) [inline sound/soc/intel/boards/cht_bsw_rt5672.c:138]: ok

File: sound/soc/intel/boards/bytcht_da7213.c (hunk 1 above)
Comment (xun2z) [reply sound/soc/intel/boards/bytcht_da7213.c]: ok, I misunderstood your request that I need to create some widgets in the kernel, while these should all be done in the sof part. I'll remove them all.

**Comment by xun2z** [comment]
I've removed all if statements and the widgets I created. Now this pull request will actually not  fix the error message until corresponding SOF topology is changed. 

By the way, in topology, I notice that there is a small inconsistent thing between the files I'm gonna modify and those already have virtual widgets on them.

For example, in 

> sof-apl-da7219.m4

[codec0_in](https://github.com/thesofproject/sof/blob/a2e6f82ef810a6a838783467472dfceb7933afc1/tools/topology/sof-apl-da7219.m4#L188)

while in 

> bytcr_rt5651.c

[codec_in0](https://github.com/thesofproject/linux/blob/2fc33699b07d18a9cf0c59b90479a3f640c1888d/sound/soc/intel/boards/bytcr_rt5651.c#L322)

I'm not sure about the difference and still checking.

File: sound/soc/intel/boards/bytcht_da7213.c [hunk 12]
```diff
@@ -56,7 +56,7 @@ static const struct snd_soc_dapm_route audio_map[] = {
 	{"MIC1", NULL, "Headset Mic"},
 	{"MIC2", NULL, "Mic"},
 
-#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL)
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL)
```
Comment (plbossart) [inline sound/soc/intel/boards/bytcht_da7213.c]: I believe I was clear that this needed to be removed?

Comment (xun2z) [reply sound/soc/intel/boards/bytcht_da7213.c]: This version is outdated.

**Comment by xun2z** [comment]
@plbossart I have rebased and updated this pr, please check.

**Comment by xun2z** [comment]
@plbossart SOF topology part is here
https://github.com/thesofproject/sof/pull/803

File: sound/soc/sof/intel/Kconfig [hunk 13]
```diff
@@ -249,4 +249,4 @@ config SND_SOC_SOF_HDA
 
 endif ## SND_SOC_SOF_INTEL_PCI
 
-endif ## SND_SOC_SOF_INTEL
+endif ## SND_SOC_SOF_INTEL_TOPLEVEL
```
Comment (plbossart) [inline sound/soc/sof/intel/Kconfig:252]: good catch, I missed this one. Thanks!

**Pull request merged**